extensions/visualization/README.md

```diff
@@ -0,0 +1,76 @@
+| vis:earth_engine_script_name  | string     | Optional URL linking to the JavaScript script file used for the Earth Engine site code snippet. |
+
+\* = One of the Image Visualization, Table Visualization, or Polygon Visualization, defined below:
+
+### Image Visualization
+
+| Field Name | Type    | Description                                            |
+|------------|---------|--------------------------------------------------------|
+| band_vis   | BandVis | Vis args for the given band_names.                     |
+| global_vis | BandVis | If vis_args are present here, they apply to all bands. |
+| image_id   | string  | Specific image to use for visualization.               |
+
+### Table Visualization
+
+| Field Name    | Type   | Description                                                                                                                                       |
+|---------------|--------|---------------------------------------------------------------------------------------------------------------------------------------------------|
+| property_name | string | Property name to select (optional)                                                                                                                |
+| color         | string | Name of a color or RGB hex code.                                                                                                                  |
+| point_size    | int    | Default size in pixels of the point markers.                                                                                                      |
+| point_shape   | string | Default shape of the marker to draw at each point location. ( TODO - list acceptable strings/find a way to express something similar to an enum ) |
+| width         | double | Default line width for lines and outlines for polygons/point shapes.                                                                              |
+| fill_color    | string | The color for filling polygons and point shapes. Defaults to 'color' at opacity 0.66.                                                             |
+
+### Polygon Visualization
+
+| Field Name    | Type    | Description                                              |
+|---------------|---------|----------------------------------------------------------|
+| property_name | string  | The property to visualize.                               |
+| property_vis  | BandVis | Parameters for the visualization of the resulting image. |
+
+BandVis is defined as follows:
+
+## Band Visualization
+
+| Field Name | Type     | Description                                                                                    |
+|------------|----------|------------------------------------------------------------------------------------------------|
+| min        | [double] | Minimum value of each band. If only one value then it is applied as the minimum for all bands. |
```

Comment:
Is there some way to express optionality? For example, in Raster Foundry land, we consider these [optional](https://github.com/raster-foundry/raster-foundry/blob/37754fc633fc3efdf8e8b6d477b75313cd812c7a/app-backend/backsplash-core/src/main/scala/com/rasterfoundry/backsplash/color/ColorCorrect.scala#L26-L27), since we assume for color correction that we have statistical information about the image, but that a user might want to override it. Does that match the intention here?

Reply:
Everything that is not required explicitly is optional.

Reply:
I mean optional in the "value or `null`" sense on the individual array items. For example, if I had some bands where I want to set a minimum, but not others, I don't believe I can express that here (unless I've gotten too used to types, and `[0, null, 0]` would be a valid value here)